### PR TITLE
chore: publish teams app in developer portal

### PR DESCRIPTION
**Pull request by yuqizhou77**
- after TDP integration feature is turned on:
image
We will have "publish in developer poral" to replace the current link to developer portal which will ask user to select env and manifest file, and then redirect user to the "publish to org" page in developer portal. The link will be something like https://dev.teams.microsoft.com/apps/{appid}/distributions/app-catalog?login_hint={account}&referrer=teamstoolkit_vsc

image

- update url handler as updating url in dev portal

### CODE REVIEW

**Review by nliu-ms — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -294,7 +294,7 @@
   "plugins.appstudio.overwriteAndUpdate": "Overwrite and update",
   "plugins.appstudio.emptyAppPackage": "Could not find any files in the package of app %s.",
   "plugins.appstudio.unprocessedFile": "%s was not processed by Teams Toolkit.",
-  "plugins.appstudio.viewDevPortal": "View in Developer Portal",
```
Comment (yuqizhou77) [inline packages/fx-core/resource/package.nls.json:297]: this has not been localized yet. So I feel it is safe to do the rename here.

File: packages/vscode-extension/package.nls.json [hunk 2]
```diff
@@ -171,6 +171,8 @@
     "teamstoolkit.commandsTreeViewProvider.publish.blockTooltip": "Cannot run command when publishing package. Try again when publish is finished.",
     "teamstoolkit.commandsTreeViewProvider.publish.running": "Publishing...",
     "teamstoolkit.commandsTreeViewProvider.publishDescription": "Publish to Teams",
+    "teamstoolkit.commandsTreeViewProvider.publishInDevPortalTitle": "Publish in Developer Portal",
+    "teamstoolkit.commandsTreeViewProvider.publishInDevPortalDescription": "Publish in Developer Portal",
```
Comment (tecton) [inline packages/vscode-extension/package.nls.json:175]: Is description same with title?

File: packages/vscode-extension/src/handlers.ts [hunk 3]
```diff
@@ -965,6 +965,13 @@ export async function publishHandler(args?: any[]): Promise<Result<null, FxError
   return await runCommand(Stage.publish);
 }
 
+export async function publishInDeveloperPortalHandler(
+  args?: any[]
+): Promise<Result<null, FxError>> {
+  ExtTelemetry.sendTelemetryEvent(TelemetryEvent.PublishStart, getTriggerFromProperty(args));
```
Comment (tecton) [inline packages/vscode-extension/src/handlers.ts:971]: Do we need to use a new event? `Publish Start` is also used by publish action. We cannot tell the difference between them.

File: packages/vscode-extension/src/treeview/webViewProvider/deploymentView.tsx [hunk 4]
```diff
@@ -3,6 +3,7 @@ import "./tree.scss";
 import { getCurrentTheme, Theme } from "./theme";
```
Comment (tecton) [inline packages/vscode-extension/src/treeview/webViewProvider/deploymentView.tsx:3]: This file is not used actually.

File: packages/vscode-extension/package.nls.json (hunk 2 above)
Comment (yuqizhou77) [reply packages/vscode-extension/package.nls.json:175]: > Is description same with title?

I would keep it same for now as this is the same as what we do for "publish to teams" item. Also double confirming with PM. Will update in another PR if we decide to change.

**Review by tecton — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**